Ndev 2603 run tests on mainnet

romanova-natasha opened the pull request.

romanova-natasha committed:
add mainnet network to envs.json

add mainnet network to envs.json

romanova-natasha committed:
add only_mainnet test mark

add only_mainnet test mark

romanova-natasha committed:
Merge branch 'develop' into NDEV-2603-run-tests-on-mainnet

romanova-natasha committed:
add use BANK_PRIVATE_KEY_MAINNET as bank account for mainnet

add use BANK_PRIVATE_KEY_MAINNET as bank account for mainnet

romanova-natasha committed:
add ETH_BANK_PRIVATE_KEY_MAINNET for eth_bank_account on mainnet run

add ETH_BANK_PRIVATE_KEY_MAINNET for eth_bank_account on mainnet run

romanova-natasha committed:
add bank's mainnet envs to test container

add bank's mainnet envs to test container

romanova-natasha committed:
select "mainnet" marked tests for mainnet run

select "mainnet" marked tests for mainnet run

kristinaNikolaevaa commented on integration/tests/basic/test_event_logs.py:
```diff
@@ -31,6 +32,7 @@ def test_non_args_event(self, event_caller_contract):
         assert event_logs[0].args == {}
         assert event_logs[0].event == "NonArgs"
 
+    @pytest.mark.mainnet
```
I think one test from this file is enough

romanova-natasha replied:
Ok, will do

romanova-natasha replied:
done

kristinaNikolaevaa commented on integration/tests/basic/test_nonce.py:
```diff
@@ -34,6 +35,7 @@ def test_get_receipt_sequence(self):
 
         self.check_transaction_list(tx_hash_list)
 
+    @pytest.mark.mainnet
```
These tests fails very often on devnet , not sure about mainnet
@gigimon what do you think?

romanova-natasha replied:
Yes, I remember now. These tests are not stable. I guess you're right we should exclude it from run. @gigimon Are you ok with it?

gigimon replied:
no, we don't need.

romanova-natasha replied:
removed

kristinaNikolaevaa commented on integration/tests/basic/test_wneon.py:
```diff
@@ -49,11 +49,12 @@ def get_balances(self, wneon, address):
         recipient_account = self.accounts[1]
         neon_balance_before, wneon_balance_before = self.get_balances(wneon, recipient_account.address)
-        deposit_amount = random.randint(1, 10)
+        deposit_amount = 0.1 #random.randint(1, 10)
```
comment

romanova-natasha replied:
thx, removed

kristinaNikolaevaa commented on envs.json:
```diff
@@ -94,6 +94,28 @@
     ]
   },
+  "mainnet": {
+    "evm_loader": "NeonVMyRX5GbCrsAHnUwx1nYYoJAtskU1bWUo6JGNyG",
```
we can add one or two tests for erc20 and many tests from rpc folder - many of them are free
also one or two tests from transfer folder

romanova-natasha replied:
Ok, will do

romanova-natasha replied:
done

kristinaNikolaevaa commented on integration/tests/basic/test_deposit.py:
```diff
@@ -127,6 +130,7 @@ def test_transfer_spl_token_from_solana_to_neon(self, solana_account, pytestconf
 
         assert int(ata_balance_after.value.amount) == int(ata_balance_before.value.amount) + full_amount
 
+    @pytest.mark.mainnet
```
tests with multipletokens mark wouldn't work on mainnet

romanova-natasha replied:
When we run tests with network == mainnet it will run only tests with mark "mainnet"
The logic you can see here https://github.com/neonlabsorg/neon-tests/blob/NDEV-2603-run-tests-on-mainnet/integration/tests/conftest.py#L51

kristinaNikolaevaa replied:
but now we have several tests with 2 marks: mainnet and multipletokens

romanova-natasha replied:
I did not include any tests with mark multipletokens for now cause we do not have multitokens feature in mainnet yet. i will doublecheck it

romanova-natasha replied:
yep, I removed these 2 tests from mainnet run, thx

romanova-natasha committed:
rm debug

rm debug

romanova-natasha committed:
add rpc tests to mainnet run

add rpc tests to mainnet run

romanova-natasha committed:
add erc20 tranfer and neon transfer tests to mainnet run

add erc20 tranfer and neon transfer tests to mainnet run

romanova-natasha committed:
rm multitoken tests from mainnet run

rm multitoken tests from mainnet run

gigimon commented on integration/tests/basic/erc/test_ERC20SPL.py:
```diff
@@ -47,6 +47,7 @@ def test_metaplex_data(self, erc20_contract):
         assert metadata["is_mutable"] is True
 
+    @pytest.mark.mainnet
     def test_balanceOf(self, erc20_contract):
```
I think we can add all read tests (test_name, test_metaplex_data)

romanova-natasha replied:
done

gigimon commented on integration/tests/basic/rpc/test_neon_rpc_base_calls.py:
```diff
@@ -63,6 +63,7 @@ def test_neon_cli_version(self, json_rpc_client):
         ), f"Version format is not correct. Pattern: {pattern}; Response: {response}"
 
+    @pytest.mark.mainnet
     def test_neon_get_solana_transaction_by_neon_transaction(self, event_caller_contract, json_rpc_client, sol_client):
```
left just one (with several transactions)

romanova-natasha replied:
done

gigimon commented on integration/tests/basic/rpc/test_rpc_base_calls.py:
```diff
@@ -183,6 +187,7 @@ def test_eth_send_raw_transaction(self, json_rpc_client):
+    @pytest.mark.mainnet
     def test_eth_sendRawTransaction_max_size(self, json_rpc_client):
         """Validate max size for transaction, 127 KB"""
         size = 127 * 1024
```
i think it's not necessary

romanova-natasha replied:
removed

gigimon commented on integration/tests/basic/rpc/test_rpc_base_calls.py:
```diff
@@ -198,6 +203,7 @@ def test_eth_sendRawTransaction_max_size(self, json_rpc_client):
         assert rpc_checks.is_hex(response["result"]), f"Invalid response result {response['result']}"
 
+    @pytest.mark.mainnet
     def test_eth_sendRawTransaction_max_contract_size(self, json_rpc_client):
```
i think it's not necessary

romanova-natasha replied:
removed

gigimon commented on integration/tests/basic/rpc/test_rpc_estimate_gas.py:
```diff
@@ -91,6 +91,7 @@ def test_eth_estimate_gas_with_big_int(self, contract_name, process_gas, reserve
         assert deploy_trx_big_gas.get("status"), f"Transaction is incomplete: {deploy_trx_big_gas}"
         assert estimated_gas >= int(deploy_trx_big_gas["gasUsed"]), "Estimated Gas < Used Gas"
 
```
I think we don't need this test file, because we will get errors here (this gas counts actual for devnet only, because mainnet has another version)

romanova-natasha replied:
removed

gigimon commented on integration/tests/basic/test_wneon.py:
```diff
@@ -95,6 +97,7 @@ def test_withdraw(self, wneon):
 
+    @pytest.mark.mainnet
     def test_transfer_and_check_token_does_not_use_spl(self, wneon):
         deposit_amount = 10
```
don't need, because we checked this contract in devnet and contract can't be changed

romanova-natasha replied:
removed

romanova-natasha committed:
deselect unnecessary tests, add eth_bank_account to erc20 classes to deposit contract from it

deselect unnecessary tests, add eth_bank_account to erc20 classes to deposit contract from it

romanova-natasha committed:
add mainnet to stand choice

add mainnet to stand choice

romanova-natasha committed:
add restore eth bank account balance after session, reduce amount of initial deposit for accounts and erc20 contracts

add restore eth bank account balance after session, reduce amount of initial deposit for accounts and erc20 contracts

romanova-natasha committed:
restore eth bak account balance for class fixture accounts

restore eth bak account balance for class fixture accounts

romanova-natasha committed:
fix test_eth_get_block_by_number_via_tags: reduce send neon amount

fix test_eth_get_block_by_number_via_tags: reduce send neon amount

romanova-natasha committed:
log accounts from which to restore eth bank account balance

log accounts from which to restore eth bank account balance

romanova-natasha committed:
restore eth bank account balance if contract balance > 2

restore eth bank account balance if contract balance > 2

romanova-natasha committed:
add: use accounts fixture for erc20 contract account

add: use accounts fixture for erc20 contract account

romanova-natasha committed:
estimate gas with multiplier 10%

estimate gas with multiplier 10%

kristinaNikolaevaa commented on integration/tests/conftest.py:
```diff
@@ -40,18 +40,23 @@ def pytest_collection_modifyitems(config, items):
+    
+    if network_name == "mainnet":
+        deselected_marks.append("only_stands")
+        deselected_marks.append("only_devnet")
```
Is it needed? I thought if you run tests with only mark "mainnet"  , all other marks automatically deselect

romanova-natasha replied:
Yes, you're right. Everything works without this marks. I deleted it

kristinaNikolaevaa commented on integration/tests/conftest.py:
```diff
@@ -196,13 +220,29 @@ def erc20_spl(
+                 eth_bank_account
+):
+    network = pytestconfig.getoption("--network")
+    if network == "mainnet":
```
you can do it without conditions, always with owner=accounts_session[0, for all networks

romanova-natasha replied:
fixed

kristinaNikolaevaa commented on utils/web3client.py:
```diff
@@ -340,6 +340,28 @@ def send_tokens(
+        self,
+        from_: eth_account.signers.local.LocalAccount,
+        to: tp.Union[str, eth_account.signers.local.LocalAccount],
+        value: int,
```
maybe better to delete value and get it here with get balance function? "all tokens" in the name with value in parameters doesn't look logic

romanova-natasha replied:
fixed

romanova-natasha committed:
fix review

fix review

romanova-natasha committed:
increase erc20 contract account deposit before tests

increase erc20 contract account deposit before tests

romanova-natasha committed:
rm unnecessary deselected marks for mainnet run

rm unnecessary deselected marks for mainnet run

romanova-natasha committed:
fix erc20 contract balance for mainnet run

fix erc20 contract balance for mainnet run

romanova-natasha committed:
fix withdraw contract for mainnet run, rm dbg print

fix withdraw contract for mainnet run, rm dbg print

romanova-natasha committed:
add info how to run tests on mannet

add info how to run tests on mannet

romanova-natasha committed:
add a cost estimation of a single mainnet test run

add a cost estimation of a single mainnet test run

romanova-natasha committed:
fix info collection for allure

fix info collection for allure
